### ISSUE TITLE
no-only-test rule should not have a "fix" 

### ISSUE THREAD

**Issue by alockwood05** [avajs/eslint-plugin-ava#189]
I imagine it is somewhat common to auto --fix code with eslint.  In which case the `t.only()` tests get fixed to `t()`, which defeats the purpose.  Can we remove this "feature"?

**Comment by novemberborn** [comment]
I've seen Atom linter plugins with options to exclude certain rules from auto fixing. Perhaps that's the way to go. What do you think @sindresorhus?

**Comment by sindresorhus** [comment]
Yes, that's what we did with the Atom XO linter: https://github.com/xojs/atom-linter-xo/blob/bb298da44f10bbc7cc52b88ce7fbb4c9f83540af/index.js#L53-L61

**Comment by revelt** [comment]
If you use Atom, you can try `linter-eslint` setting:
<img width="411" alt="atom" src="https://user-images.githubusercontent.com/8344688/41095443-5a49825a-6a49-11e8-94cb-8e829df4a026.png">

I got this setup working with Prettier piggybacking on top of ESLint as well. FIY, if you use standalone Prettier you [won't be able](https://github.com/prettier/prettier-atom/issues/413) to exclude "fixing".

I disagree about rationality of proposed solution to "remove feature" — there is dev scenario when you want "fix" to be flagged up but not to work. In prod scenario, when running unit tests in CI for example, you want "fix" to be enabled and work. I think _removing_ the feature would be a bad idea. Just disable the auto-fix!

**Comment by hbarcelos** [comment]
Couldn't this rule have support to be disabled somehow in the `.eslintrc` file?

Something like:
```json
    "ava/no-only-test": ["warn", {
      "autofix": false
    }]
```

**Comment by alockwood05** [comment]
Sounds good to me.  See fix making sense for prod environments. I believe a solution to this is to set the rule to `"warn"`  and and use eslint options `--fix --quiet`.  Which I understand will not fix the warnings.
